docs/styleguides/python.md

```diff
@@ -0,0 +1,259 @@
+# DC/OS Python Style Guide
+###### Version 0.1
+###### April-6-2017
+
+For Python source code within DC/OS project, we prefer to keep [Google Python Style Guide Revision 2.59] as the base and
+**extend** it with some changes that will make DC/OS codebase look modern, consistent and easy to follow.
+
+##### Rationale
+
+With any style guide, the primary requirement is to adopt one and follow it consistently. With our choices below, we
+can use tools available in the python ecosystem to our advantage, enforce consistency in our codebase, while creating
+and publishing automatically generated documentation from the code itself.
+
+##### Versioned document
+
+Styleguide _conventions_ are not set in stone. We can improve upon this or remove inconvenient styles by creating a PR,
+discussing the merits or demerits to arrive at a consensus. Developers who contribute code to the DC/OS repo are the
+main stake holders of this style guide.
+
+##### Python Version Requirement
+
+This style guide is written for Python3.5 and above. As new features of the language get standardized, this style guide
+can be revised to adopt new features.
+
+#### Base Style Guide
+
+* [Google Python Style Guide Revision 2.59]
+
+
+#### DC/OS Specific extensions to Base Style Guide
+
+* Import Requirements
+    * Adopt the base style guide preferences.
+
+        ```
+        Use import x for importing packages and modules.
+        Use from x import y where x is the package prefix and y is the module name with no prefix.
+        Use from x import y as z if two modules named y are to be imported or if y is an inconveniently long name.
+        ```
+    * Use `__init__.py` only to represent python packages.
+        * Prefer empty `__init__.py` modules. Keep definitions in submodules.
+
+    _Rationale_: Keeping `__init__.py` empty helps in avoid unnecessary imports when you import a submodule from
+    package. Plus,it helps in navigating the code base you know that an object is strictly defined in a submodule and
+    not aliased in `__init__.py`
+
+* Import formatting
+
+    * The types of imports should be separated by single line and order should be.
+
+        1. standard library imports.
+        2. third-party imports.
+        3. application specific imports.
+
+    * The `from` style import and `import` statements should be separated by single line too.
+    * Each group of these imports should be sorted.
+
+    _Rationale_:  Readablity and code-foldability are reasons for this separation.
```

Comment:
> Yes, I am referring to folding features available in all editors. "zf", "zo", "za" in vim.
And [+] in IntelliJ.

What if I somebody does not use it/is a die-hard purist and uses plain vim? Does it mean that this rule does not concern him?

> A good reasoning to keep from and import statements separate are they are two different types of statements. The similar members of the type can be grouped together. My argument is, this grouping and separation within groups helps in readability. This separation also helps editors with code-folding and auto-imports features.

I think this is already achieved by sorting your imports correctly:
```
import a.b
import c
from d import z
from e import x
```
I think it's easy to distinguish within a particular group `import` subgroup and `from` subgroup.

> True. What I've seen this folks using # comments to describe or distinguishing it based on familiarity.

Are you suggesting that we start commenting as well if there is ambiguity?